bestbot.py

```diff
@@ -414,6 +422,74 @@ async def pls(context, animal = None, noarg = None):
     else:
         await context.send(f'{context.author.mention} {ERROR_REPLY}.')
 
+@client.command(brief       = 'Allows a user to input a game with its activation code', ######################## addgame
+                description = 'Allows a user to input a game with its activation code. arg1 = gameName, arg2 = code.')
+async def addgame(context, gameName = None, gameCode = None, noarg = None):
+    try:
+        if noarg is not None or\
+            gameName is None or\
+            gameCode is None:
+                raise Exception()
+    except Exception:
+        await context.send(f'{context.author.mention} {ERROR_REPLY}.')
+        return
+    
+    with open(files["f_gameKeys"], "r") as gameKeyFile:
+        games = json.load(gameKeyFile)
+
+    games[str(gameName)] = str(gameCode)
+
+    with open(files["f_gameKeys"], "w") as gameKeyFile:
+        games = json.dump(games,gameKeyFile)

```

Comment:
```suggestion
    with open(files["f_gameKeys"], "r+") as gameKeyFile:
        games = json.load(gameKeyFile)
			games[str(gameName)] = str(gameCode)
			json.dump(games, gameKeyFile)
```
Not tested, but shouldn't this work? We can open a file in both read and write in one go, and then do what we need. Additionally we don't need to store the dump result as a variable as it isn't used any more